source/includes/usage-examples/code-snippets/UpdateOne.java

```diff
@@ -0,0 +1,46 @@
+// ignored first line
+package usage.examples;
+
+import org.bson.Document;
+import org.bson.conversions.Bson;
+
+import com.mongodb.MongoException;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.UpdateOptions;
+import com.mongodb.client.model.Updates;
+import com.mongodb.client.result.UpdateResult;
+
+public class UpdateOne {
+
+    public static void main(String[] args) {
+        // Replace the uri string with your MongoDB deployment's connection string
+        String uri = "mongodb+srv://<user>:<password>@<cluster-url>?retryWrites=true&w=majority";
+        MongoClient mongoClient = MongoClients.create(uri);
+
+        MongoDatabase database = mongoClient.getDatabase("sample_mflix");
+        MongoCollection<Document> collection = database.getCollection("movies");
+
+        try {
+            Document query = new Document().append("title",  "Cool Runnings");
+
+            Bson updates = Updates.combine(
+                    Updates.set("runtime", 99),
+                    Updates.addToSet("genres", "Sports"),
+                    Updates.currentTimestamp("lastUpdated"));
+
+            UpdateOptions options = new UpdateOptions().upsert(true);
```

Comment:
If we're only doing one example, perhaps upsert is not the most common use case.  Do we have any research suggesting otherwise?

Reply:
In particular, I can't see a real movie database application being designed like this.  You would typically know in advance whether the database contained a document for the movie you want to add a genre to.

Reply:
The usage examples are admittedly contrived; I wanted to show how to set the UpdateOptions. Alternatively, I could add a short snippet following the paragraph about optional parameters that shows an instance of UpdateOptions being created and then added into a query. 

The Node.js example for updateOne is somewhat similar to this one -- the update, specifying upsert, sets the "plot" field of a movie. I could change this to use the "theaters" collection with an updated location if that seems more appropriate to you. Let me know which combination of changes works best for you.